lrlex/src/lib/lexer.rs

```diff
@@ -201,7 +202,7 @@ impl<'a, StorageT: Copy + Eq + Hash + PrimInt + Unsigned> Lexer<StorageT>
                 }
             }
             if longest > 0 {
-                self.newlines.extend(
+                unsafe { &mut *self.newlines.get() }.extend(
```

Comment:
Why is this unsafe?

Reply:
It's an `UnsafeCell`.

Reply:
Why does `newlines` need to be in an `UnsafeCell`?

Reply:
The commit message https://github.com/softdevteam/grmtools/pull/120/commits/52351a708f41150079754b611f601e038bbeda67 hopefully helps. Basically removing `&mut self` means we have to move to interior mutability and for something like a `Vec` that means `UnsafeCell` (with all its attendant horrors) is our only option.

Reply:
I think I understand now.